src/app/(admin)/admin/_components/ServiceProvider/SocialLinks.js

```diff
@@ -1,17 +1,43 @@
 import React from 'react';
 import { FormFieldWrapper } from '@admin/components/FormFieldWrapper';
 import { SpecialistFormFields, SpecialistFormSections } from '@admin/_lib/specialistData';
-import { TextInputList } from '@admin/components/TextInputList';
 import PropTypes from 'prop-types';
+import { required, TextInput } from 'react-admin';
+import { useFormState, useWatch } from 'react-hook-form';
+import { MAX_NUM_SELECTED_SOCIAL_LINKS } from '@admin/_lib/consts';
 
 export function SocialLinks({ className }) {
   const { instagram, facebook, youtube, linkedin, tiktok, viber, telegram } = SpecialistFormFields;
   const socialMediaList = [instagram, facebook, youtube, linkedin, tiktok, viber, telegram];
 
+  const socialLinkObject = useWatch({ name: 'socialLink' });
+  const selectedSocialLinkNum = socialLinkObject ? Object.values(socialLinkObject).filter(link => link)?.length : 0;
+
+  const { errors } = useFormState();
+
+  const isSelectedLinkNumWithinAllowedRange = selectedSocialLinkNum <= MAX_NUM_SELECTED_SOCIAL_LINKS;
+  const errorMessage = errors?.socialLink ? errors?.socialLink.root.message : '';
+
+  const displayError = !isSelectedLinkNumWithinAllowedRange && (
+    <p className="my-1 text-c3 tracking-wide text-system-error">{errorMessage}</p>
+  );
```

Comment:
It is not obvious that isSelectedLinkNumWithinAllowedRange and errorMessage have a relation, errorMessage may contain an error not related to the maximum number of links, so either replace !isSelectedLinkNumWithinAllowedRange && with errorMessage &&, or replace errorMessage with a text value, e.g "You cannot add more than 5 social media links". I prefer the first option

Reply:
The idea of using additional variable isSelectedLinkNumWithinAllowedRangeis was to hide error message without re-rendering the page once an admin fixes it and keeps the amount of social links within allowed range. If display error is shown based on the errorMessage, to hide it on the UI will require a user to click on SAVE button which might be a little confusing. 

Even though the amount of social links should not trigger any errors, the error is still there and will be re-validated once used clicks on the SAVE button.
![image](https://github.com/keenethics/mayak/assets/103147058/4311c63b-06aa-45bf-a06b-421c97fd498e)